src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/PropertiesShouldNotBeWriteOnly.cs

```diff
@@ -57,40 +57,41 @@ public override void Initialize(AnalysisContext context)
         /// </summary>
         private static void AnalyzeSymbol(SymbolAnalysisContext context)
         {
-            if (context.Symbol is not IPropertySymbol property)
-            {
-                return;
-            }
+            var property = (IPropertySymbol)context.Symbol;
 
             // not raising a violation for when:
             //     property is overridden because the issue can only be fixed in the base type
             //     property is the implementation of any interface member
-            if (property.IsOverride || property.IsImplementationOfAnyInterfaceMember())
+            if (property.IsOverride || GetRule(property) is not DiagnosticDescriptor descriptor || property.IsImplementationOfAnyInterfaceMember())
```

Comment:
**NOTE:** The checks in `GetRule` were previously done **after** calling `IsImplementationOfAnyInterfaceMember`.

I think that checks in `GetRule` are a lot cheaper and can eliminate lots of properties. So this might slightly have a performance impact. This analyzer is generally fast enough though.